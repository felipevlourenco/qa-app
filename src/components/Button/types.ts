import { ReactNode } from 'react'

export interface ButtonProp {
  children: ReactNode
  onClick?: () => void
  disabled?: boolean
  variant?: 'primary' | 'secondary'
  type?: 'button' | 'submit' | 'reset'
}
