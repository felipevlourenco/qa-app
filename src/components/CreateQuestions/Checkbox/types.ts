export interface CheckboxProp {
  checked: boolean
  onChange: () => void
}
