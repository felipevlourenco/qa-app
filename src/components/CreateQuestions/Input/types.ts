import { FieldError, UseFormRegister } from 'react-hook-form'
import { AddQuestionAction } from 'store/questions/types'

export interface InputProp {
  id: 'question' | 'answer'
  label: string
  defaultValue?: string
  register: UseFormRegister<AddQuestionAction>
  errors: {
    question?: FieldError | undefined
    answer?: FieldError | undefined
  }
}
