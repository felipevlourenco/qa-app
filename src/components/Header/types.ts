import { ReactNode } from 'react'

export interface HeaderProp {
  children: ReactNode
}
