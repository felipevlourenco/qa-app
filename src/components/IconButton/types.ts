export interface IconButtonProp {
  icon: string
  alt: string
  onClick: () => void
}
