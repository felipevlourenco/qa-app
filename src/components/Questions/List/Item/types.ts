import { Question } from 'store/questions/types'

export interface QuestionItemProp {
  question: Question
}
