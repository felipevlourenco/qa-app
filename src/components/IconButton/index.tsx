import React from 'react'
import { IconButtonProp } from 'components/IconButton/types'
import styles from 'components/IconButton/styles.module.css'

const IconButton = ({ icon, onClick, alt }: IconButtonProp) => {
  return (
    <button type="button" onClick={onClick} className={styles.button}>
      <img src={icon} alt={alt} className={styles.img} />
    </button>
  )
}

export default IconButton
