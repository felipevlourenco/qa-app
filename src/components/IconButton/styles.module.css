.button {
  background-color: transparent;
  cursor: pointer;
  border: 0;
}

.img {
  width: 20px;
  height: 20px;
  padding: 0.5rem;
}

.img:hover {
  border-radius: 4px;
  box-shadow: 5px 5px 5px #00000020;
}
