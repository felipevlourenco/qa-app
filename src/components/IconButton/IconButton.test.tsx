import { fireEvent, render } from '@testing-library/react'
import IconButton from 'components/IconButton'
import { IconButtonProp } from 'components/IconButton/types'
import editIcon from 'assets/edit.svg'

const defaultProp: IconButtonProp = {
  onClick: jest.fn,
  icon: editIcon,
  alt: 'icon'
}

describe('components/IconButton', () => {
  it('should render', () => {
    const { container } = render(<IconButton {...defaultProp} />)
    expect(container).toMatchSnapshot()
  })

  it('should call onClick callback', () => {
    const onClick = jest.fn()
    const { getByRole } = render(<IconButton {...defaultProp} onClick={onClick} />)
    fireEvent.click(getByRole('button'))
    expect(onClick).toBeCalledTimes(1)
  })
})
