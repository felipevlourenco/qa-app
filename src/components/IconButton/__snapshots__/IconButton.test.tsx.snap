// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IconButton should render 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    <img
      alt="icon"
      class="img"
      src="edit.svg"
    />
  </button>
</div>
`;
